p2p/src/lib.rs

```diff
@@ -156,6 +151,33 @@ impl subsystem::Subsystem for Box<dyn P2pInterface> {}
 
 pub type P2pHandle = subsystem::Handle<Box<dyn P2pInterface>>;
 
+pub type P2pNetworkingService = DefaultNetworkingService<NoiseTcpTransport>;
+
+pub fn make_p2p_transport() -> NoiseTcpTransport {
+    let stream_adapter = NoiseEncryptionAdapter::gen_new();
+    let base_transport = net::default_backend::transport::TcpTransportSocket::new();
+    NoiseTcpTransport::new(stream_adapter, base_transport)
+}
+
+fn get_p2p_bind_addresses(bind_addresses: &[String], p2p_port: u16) -> Result<Vec<SocketAddr>> {
```

Comment:
A better signature here would be

```
fn get_p2p_bind_addresses<S: AsRef<str>>(bind_addresses: &[S], ...)
```

Then add `as_ref()` to addresses elements, and instead of `clone()`, use `to_owned()`.

Reply:
Updated, thanks.